### ISSUE TITLE
nullpointerexception at https://github.com/elasticsearch/elasticsearch/blob/v1.1.1/src/main/java/org/elasticsearch/percolator/PercolatorService.java#L301

### ISSUE THREAD

**Issue by OlegYch** [elastic/elasticsearch#6172]
for this kind of query
```
GET /idx/type/5359b8d10cf2f2ff4ec71ed7/_percolate
{
  "filter": {
    "or": {
      "filters": [
        {}
      ]
    }
  }
}
```

**Comment by clintongormley** [comment]
This is still broken in 2.0

**Comment by qwerty4030** [comment]
@clintongormley I took a look and seems like this is a fairly easy fix. I'm assuming the expected behavior is to ignore the `filter` in this case (all percolators executed). I wanted to add an integration test and noticed the [PercolateRequestBuilder](https://github.com/elastic/elasticsearch/blob/2.x/core/src/main/java/org/elasticsearch/action/percolate/PercolateRequestBuilder.java) and [PercolateSourceBuilder](https://github.com/elastic/elasticsearch/blob/2.x/core/src/main/java/org/elasticsearch/action/percolate/PercolateSourceBuilder.java) had the `filter` option removed in this commit: [Query DSL: Remove filter parsers.](https://github.com/elastic/elasticsearch/commit/a0af88e99630b9d3f0c2cf4997f2e82f1f834d41) However the [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/2.3/search-percolate.html#_percolate_api) indicates both the `query` and `filter` options are still supported. My plan was to add the `filter` option back to those java client classes as a `QueryBuilder`. Let me know if I'm not on the right track here. Thanks.

**Comment by martijnvg** [comment]
Hey @qwerty4030 the best way to fix this NPE is to replace line 321 in `PercolatorService.java` (2.x branch) with this:

```java
ParsedQuery parsedQuery = documentIndexService.queryParserService().parseInnerFilter(parser);
if (parsedQuery != null) {
   context.percolateQuery(new ConstantScoreQuery(parsedQuery.query()));
}
```

Filters are deprecated and therefor adding filter methods to the Java api isn't a good idea.
This issue doesn't occur in the master branch (percolator has been rewritten), but it makes sense to fix it in 2.x and 2.3 branches. If you open a PR for this then I'm happy to merge it.

**Comment by martijnvg** [comment]
Fixed via #18563
